.github/workflows/go.yml

```diff
@@ -59,8 +106,140 @@ jobs:
+
+    - name: Setup MINGW32
+      if: matrix.PLATFORM_ID == 'x86'
+      uses: msys2/setup-msys2@v2
+      with:
+        msystem: MINGW32
+        path-type: inherit
+        install: mingw-w64-i686-gcc
+
+    - name: Setup golang 386, Setup Environment
+      if: matrix.PLATFORM_ID == 'x86'
+      uses: ./.github/workflows/386-environment
+
+    - name: Setup golang x64
+      if: matrix.PLATFORM_ID == 'x64'
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14
+
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-${{ matrix.PLATFORM_ID }}-go-${{ hashFiles('**\go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.PLATFORM_ID }}-go-${{ hashFiles('**\go.sum') }}
+
+    - name: Build mackerel-agent
+      run: |
+        $COMMIT = git rev-parse --short HEAD
+        mkdir build/
+        go build -o build/mackerel-agent.exe -ldflags="-X main.gitcommit=$COMMIT" github.com/mackerelio/mackerel-agent
+        go build -o build/mackerel-agent-kcps.exe -ldflags="-X main.gitcommit=$COMMIT -X github.com/mackerelio/mackerel-agent/config.apibase=http://198.18.0.16" github.com/mackerelio/mackerel-agent
+      env:
+        GOARCH: ${{ matrix.GOARCH }}
+        CC: ${{ matrix.CC }}
+
+    - name: Build mackerel-agent-plugins
+      run: |
+        foreach ($line in (Get-Content wix/pluginlist.txt) -as [string[]]) {
+          $piece = $line -split "/"
+          $outputFilename = $piece[$piece.Length - 1]
+          Write-Host $outputFilename
+          go build -o build/$outputFilename.exe $line
+        }
+      env:
+        GOARCH: ${{ matrix.GOARCH }}
+        CC: ${{ matrix.CC }}
+
+    - name: Build tools
+      run: |
+        go build -o build/wrapper.exe wix/wrapper/wrapper_windows.go wix/wrapper/install.go
+        go build -o build/replace.exe wix/replace/replace_windows.go wix/replace/shell_windows.go
+        go build -o build/generate_wxs.exe wix/generate_wxs/generate_wxs.go
+      env:
+        GOARCH: ${{ matrix.GOARCH }}
+        CC: ${{ matrix.CC }}
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: windows-build-artifacts-${{ matrix.PLATFORM_ID }}
+        path: |
+          build/
+
+  windows-installer:
+    name: Make Installer (Windows)
+    runs-on: windows-2019
+    needs: build-windows
+    if: github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        include:
+          - PLATFORM_ID: x86
+          - PLATFORM_ID: x64
+            MSI_SUFFIX: -x64
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Unshallow
+      run: git fetch --prune --unshallow --tags
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: windows-build-artifacts-${{ matrix.PLATFORM_ID }}
+        path: build/
+
+    - name: Build Installer
+      run: |
+        $VERSION = git describe --tags --abbrev=0 --match "v*" | %{ $_ -replace "v","" }
```

Comment:
I think this does not work as expected.
- Currently, version tag was created via GitHub Release by [`mackerel-github-release` command](https://github.com/mackerelio/golib/blob/master/cmd/mackerel-github-release/main.go#L128), invoked in ["Release to GitHub Releases" workflow job](https://github.com/mackerelio/mackerel-agent/blob/master/.github/workflows/go.yml#L85)
- This job runs BEFORE "Release to GitHub Releases" job, so desired (new) version tag does not exist

Actually, current workflow runs as following (complicated :brain:)
- Version bump p-r merged (like #682, here we assume 0.70.3 => 0.71.0)
- On this merge commit at master branch:
  - GitHub Actions runs and
    - build Linux v0.71.0 artifacts
    - creates "v0.71.0" GitHub Releases and git tags
    - upload Linux artifacts to "v0.71.0" GitHub releases
  - AppVeyor tests and builds "v0.70.3", but does not upload releases
    - https://ci.appveyor.com/project/mackerelio/mackerel-agent/builds/36818983
    - At this timing, latest git tag is still "v0.70.3"
- On "v0.71.0" tags pushed:
  - GitHub Actions does nothing since conditions are not met https://github.com/mackerelio/mackerel-agent/blob/master/.github/workflows/go.yml#L3-L6
  - AppVeyor tests and builds "v0.71.0", then upload to (pre-existing) "v0.71.0" GitHub release
    - https://ci.appveyor.com/project/mackerelio/mackerel-agent/builds/36819108

Reply:
Thank you for the detailed explanation. 🙏 
I found an implementation that used a utility (gobump) to determine the settings later in the process.
I rewrote it using it. refer d89b058